tests/pine.spec.coffee

```diff
@@ -16,23 +15,23 @@ if IS_BROWSER
 	realFetchModule = require('fetch-ponyfill')({ Promise })
 	_.assign(global, _.pick(realFetchModule, 'Headers', 'Request', 'Response'))
 else
-	settings = require('resin-settings-client')
-	apiUrl = settings.get('apiUrl')
-	dataDirectory = settings.get('dataDirectory')
+	temp = require('temp').track()
+	dataDirectory = temp.mkdirSync()
 
 fetchMock = require('fetch-mock').sandbox(Promise)
 # Promise sandbox config needs a little help. See:
 # https://github.com/wheresrhys/fetch-mock/issues/159#issuecomment-268249788
 fetchMock.fetchMock.Promise = Promise
 require('resin-request/build/utils').fetch = fetchMock.fetchMock # Can become just fetchMock after issue above is fixed.
 
-token = getToken({ dataDirectory })
+token = require('resin-token')({ dataDirectory })
+request = require('resin-request')({ dataDirectory })
```

Comment:
As we already have the resin-request conversion to DI ready I'd bring it here and merge after that, would also serve as an integration test. Other than that I have no objections so approving this PR now.